tests/providers/amazon/aws/hooks/test_s3.py

```diff
@@ -1047,6 +1047,33 @@ def test_download_file_with_preserve_name_file_already_exists(self, tmp_path):
                 use_autogenerated_subdir=False,
             )
 
+    @mock.patch.object(S3Hook, "get_session")
+    def test_download_file_with_extra_args_sanitizes_values(self, mock_session):
+        bucket = "test_bucket"
+        s3_key = "test_key"
+        encryption_key = "abcd123"
+
+        s3_hook = S3Hook(
+            extra_args={
+                "SSECustomerKey": encryption_key,
+                "SSECustomerAlgorithm": "AES256",
+                "invalid_arg": "should be dropped",
+            }
+        )
+
+        mock_obj = Mock()
+        mock_resource = Mock()
+        mock_resource.resource.return_value = mock_obj
+        mock_session.return_value = mock_resource
+
+        s3_hook.download_file(key=s3_key, bucket_name=bucket)
+
+        mock_obj.load.assert_called_once_with(
+            key=s3_key,
+            bucket_name=bucket,
+            extra_args={"SSECustomerKey": encryption_key, "SSECustomerAlgorithm": "AES256"},
+        )
```

Comment:
```suggestion
        mock_obj = Mock(name="MockedS3Object")
        mock_resource = Mock(name="MockedBoto3Resource")
        mock_resource.return_value.Object = mock_obj
        mock_session.return_value.resource = mock_resource

        s3_hook.download_file(key=s3_key, bucket_name=bucket)

        mock_obj.assert_called_once_with(bucket, s3_key)
        mock_obj.return_value.load.assert_called_once_with(
            SSECustomerKey=encryption_key, SSECustomerAlgorithm="AES256"
        )
```

Reply:
This one should fix

---

This part only for help in debugger
```python
        mock_obj = Mock(name="MockedS3Object")
        mock_resource = Mock(name="MockedBoto3Resource")
```

![image](https://github.com/apache/airflow/assets/3998685/9bb33c88-7573-421b-95be-ece29f171c37)

Reply:
That did it.   I wrestled with that one for too long, thanks.

Reply:
Pushing that change in a new commit, added you as co-author for it. :+1: